Changelog for WPCS 2.2.0 (#1833)
WordPress/WordPress-Coding-Standards: CHANGELOG.md

- The `sanitize_key()` and the `highlight_string()` functions to the `escapingFunctions` list used by the `WordPress.Security.EscapeOutput` sniff.
- The `RECOVERY_MODE_COOKIE` constant to the list of WP Core constants which may be defined by plugins and themes and therefore don't need to be prefixed (`WordPress.NamingConventions.PrefixAllGlobals`).
- `$content_width`, `$plugin`, `$mu_plugin` and `$network_plugin` to the list of WP globals which is used by both the `WordPress.Variables.GlobalVariables` and the `WordPress.NamingConventions.PrefixAllGlobals` sniffs.
- `Sniff::is_short_list()` utility method to determine whether a _short array_ open/close token actually represents a PHP 7.1+ short list.
- `Sniff::find_list_open_close()` utility method to find the opener and closer for `list()` constructs, including short lists.
- `Sniff::get_list_variables()` utility method which will retrieve an array with the token pointers to the variables which are being assigned to in a `list()` construct. Includes support for short lists.

> - `Sniff::is_function_deprecated()` `static` utility method to determine whether a declared function has been marked as deprecated in the function docblock.

Reviewer: ```suggestion
- `Sniff::is_function_deprecated()` utility method to determine whether a declared function has been marked as deprecated in the function docblock.
```

The three lines above don't highlight themselves as static methods, so there's no reason for this one too.

Reviewer: ```suggestion
- `Sniff::is_function_deprecated()` `static` utility method to determine whether a declared function has been marked as deprecated in the function DocBlock.
```

A DocBlock is stylised with capital D and capital B: https://docs.phpdoc.org/guides/docblocks.html

Author: > The three lines above don't highlight themselves as static methods, so there's no reason for this one too.

The other three methods _aren't_ static, while this one is. That's the point.

Reviewer: Can we at least make it 

> static

rather than

> `static`

There's no reason for it to be marked as code in this sentence.

Author: Done

    This documentation can be exposed via the [`PHP_CodeSniffer` `--generator=...` command-line argument](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Usage).

### Changed
- The default value for `minimum_supported_wp_version`, as used by a [number of sniffs detecting usage of deprecated WP features](https://github.com/WordPress/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#minimum-wp-version-to-check-for-usage-of-deprecated-functions-classes-and-function-parameters), has been updated to `5.0`.

> - The `WordPress.Arrays.ArrayKeySpacingRestrictions` sniff has two new errorcodes: `TooMuchSpaceBeforeKey` and `TooMuchSpaceAfterKey`. Both auto-fixable.

Reviewer 2: ```suggestion
- The `WordPress.Arrays.ArrayKeySpacingRestrictions` sniff has two new error codes: `TooMuchSpaceBeforeKey` and `TooMuchSpaceAfterKey`. Both auto-fixable.
```

- The processing of the `WordPress` ruleset by `PHP_CodeSniffer` will now be faster.
- Various minor code tweaks and clean up.
- Various minor documentation fixes.
- Documentation: updated the repo URL in all relevant places.

### Deprecated
- The `WordPress.WP.TimezoneChange` sniff. Use the `WordPress.DateTime.RestrictedFunctions` instead.
    The deprecated sniff will be removed in WPCS 3.0.0.

### Fixed
- All sniffs in the `WordPress.Arrays` category will no longer treat _short lists_ as if they were a short array.
- The `WordPress.NamingConventions.ValidFunctionName` and the `WordPress.NamingConventions.PrefixAllGlobals` sniff will now ignore functions marked as `@deprecated`.
- Both the `WordPress.NamingConventions.PrefixAllGlobals` sniff as well as the `WordPress.WP.GlobalVariablesOverride` sniff have been updated to recognize variables being declared via (long/short) `list()` constructs and handle them correctly.
- Both the `WordPress.NamingConventions.PrefixAllGlobals` sniff as well as the `WordPress.WP.GlobalVariablesOverride` sniff will now take a limited list of WP global variables _which are intended to be overwritten by plugins/themes_ into account.

>     Initially this list contains the `$content_width` and the `$wp_cockneyreplace` variables.

Reviewer 2: ```suggestion
    Initially this list contained the `$content_width` and the `$wp_cockneyreplace` variables.
```

Author: @GaryJones Second opinion please.

IMO the present tense is correct as the list has not changed since the fix this changelog entry refers to, was pulled and merged and this is the first mention of this list in the changelog.

Reviewer: Agreed - `contains`, as it still talking in the present - "...sniff will NOW take a limited list...[and that list] contains...".

> - `WordPress.NamingConventions.ValidHookName`: will no longer examine a string array access index key as it it were part of the hook name.

Reviewer 2: ```suggestion
- `WordPress.NamingConventions.ValidHookName`: will no longer examine a string array access index key as if it were part of the hook name.
```

Author: @GaryJones Second opinion please. AFAICS the original sentence is correct English.

Reviewer 3: As far as I know both were/was are valid here and it would be a matter of personal preference. I read this aloud about 20 times and it felt equally acceptable saying either version.

Reviewer: Third person singular (he, she, it), subjunctive mood (imaginary, hypothetical) of "to be", means it should be:

```
- `WordPress.NamingConventions.ValidHookName`: will no longer examine a string array access index key as if it were a part of the hook name.
```

Reviewer 3: I did consider the subjunctive here but it's also indicative and thus `was` becomes valid. For me I have no preference either way, both suited me :)

Author: This should be fixed now.
